paimon-core/src/main/java/org/apache/paimon/table/Table.java

```diff
@@ -58,6 +58,8 @@ public interface Table extends Serializable {
     /** Optional comment of this table. */
     Optional<String> comment();
 
+    boolean isFileStoreTable();
```

Comment:
`Table` is the top level API. Regarding the write path, maybe we need a new design that doesn't need to be exposed here yet.
I think we can not expose this method now. We can just try cast to `FileStoreTable`.